hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -2322,21 +2322,22 @@ void sortDatanodes(String clientMachine, OmKeyInfo... keyInfos) {
         }
         for (OmKeyLocationInfo k : key.getLocationList()) {
           Pipeline pipeline = k.getPipeline();
-          PipelineID pipelineId = pipeline.getId();
-          List<DatanodeDetails> sortedNodes = sortedPipelines.get(pipelineId);
+          List<DatanodeDetails> nodes = pipeline.getNodes();
+          List<String> uuidList = toNodeUuid(nodes);
+          Set<String> uuidSet = new HashSet<>(uuidList);
```

Comment:
Do you think it would be better to return a Set<String> directly in toNodeUuid(...), rather than return a list and then create a set from it? Its a tiny optimisation but I don't think we really need the intermediate list - its just used in the log message later and we can probably just use "nodes" rather than nodeList in the log as it was before.

Reply:
I tried that first.  The problem is we'd need to return a sorted set (for setting up reproducible expectations in unit tests), which somewhat goes against the goal of optimization.

Reply:
OK - thanks for the explanation. We can keep it how it is then.